geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -122,21 +123,26 @@ static IProcess groundSkyViewFactor() {
                 String baseName = "rsu_ground_sky_view_factor"
                 String outputTableName = prefixName + "_" + baseName
 
-                // Other local variables
-                // Size of the grid mesh used to sample each RSU (according to the point density, we take a factor 10
-                // in order to consider that for some LCZ(7), 90% of the surfaces may be covered by buildings and
-                // then will not be used for the SVF calculation)
-                def gms = Math.pow((1.0/pointDensity)/10,0.5)
-
                 // Create the needed index on input tables and the table that will contain the SVF calculation points
-                datasource.execute(("CREATE INDEX IF NOT EXISTS ids_inI ON $rsuTable($geometricColumnRsu) USING RTREE; "+
-                        "CREATE INDEX IF NOT EXISTS ids_inA ON $correlationBuildingTable($geometricColumnBu) USING RTREE; "+
-                        "CREATE INDEX IF NOT EXISTS id_inA ON $correlationBuildingTable($idColumnRsu); "+
-                        "DROP TABLE IF EXISTS $ptsRSUtot; CREATE TABLE $ptsRSUtot (pk serial, "+
+                H2gisSpatialTable rsuSpatialTable = datasource.getSpatialTable(rsuTable)
+                H2gisSpatialTable buildingSpatialTable = datasource.getSpatialTable(correlationBuildingTable)
+                if(!rsuSpatialTable[geometricColumnRsu].spatialIndexed){rsuSpatialTable[geometricColumnRsu].createSpatialIndex()}
```

Comment:
You can just write `rsuSpatialTable[geometricColumnRsu].createSpatialIndex()` because indexes are created only if there isn't already an index (as you can see [here](https://github.com/orbisgis/orbisdata/blob/master/data-manager/src/main/java/org/orbisgis/datamanager/JdbcColumn.java#L168)).

Reply:
OK I was not sure